pkg/api/port_inspect.go

```diff
@@ -0,0 +1,34 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type PortInspectParams struct {
+	TargetHost  string `json:"target_host,omitempty"`
+	PortRange   string `json:"port_range,omitempty"`
+	ShallowScan bool   `json:"shallow_scan,omitempty"`
+}
+
+func PortInspect(request *http.Request) (interface{}, error) {
+	var (
+		portInspectParams PortInspectParams
+		data              interface{}
+	)
+
+	if err := json.NewDecoder(request.Body).Decode(&portInspectParams); err != nil {
+		return nil, err
+	}
+
+	// place for executor
+	// Executor.execute // or something
+	output_mock := []byte("{\"foo\": \"bar\"}")
+	// return result to client
```

Comment:
Copying comment from the other PR:

Now we have the `executor` package that can be used here (instead of the mocking).